fhir-cql-rest/src/main/java/com/ibm/fhir/cql/engine/rest/terminology/RestFhirTerminologyProvider.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.cql.engine.rest.terminology;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.terminology.CodeSystemInfo;
+import org.opencds.cqf.cql.engine.terminology.TerminologyProvider;
+import org.opencds.cqf.cql.engine.terminology.ValueSetInfo;
+
+import com.ibm.fhir.client.FHIRClient;
+import com.ibm.fhir.client.FHIRParameters;
+import com.ibm.fhir.client.FHIRResponse;
+import com.ibm.fhir.cql.engine.util.FHIRClientUtil;
+import com.ibm.fhir.model.resource.Bundle;
+import com.ibm.fhir.model.resource.Parameters;
+import com.ibm.fhir.model.resource.Parameters.Parameter;
+import com.ibm.fhir.model.resource.ValueSet;
+import com.ibm.fhir.model.type.code.BundleType;
+import com.ibm.fhir.model.type.code.ResourceType;
+
+public class RestFhirTerminologyProvider implements TerminologyProvider {
+
+    private static final String URN_UUID = "urn:uuid:";
+    private static final String URN_OID = "urn:oid:";
```

Comment:
We could probably push this down into ModelSupport?  @JohnTimm what do you think? it's pretty common.

Reply:
Or into your Constants.java class